### PR TITLE
add python solution for My Cow Ate My Homework (762)

### PR DESCRIPTION
**Pull request by jjliewie**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by jjliewie** [comment]
no way it worked

File: solutions/usaco-762.mdx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+n = int(input())
+
+questions = [*map(int, input().split())]
+
+left, right, total, minimum, ans = -1, n, [0] * n, [0] * n, []
+
+total[n - 1], minimum[n - 1] = questions[n - 1], questions[n - 1]
+
+for i in range(n - 2, 0, -1):
+
```
Comment (ryanchou-dev) [inline solutions/usaco-762.mdx:35]:
```suggestion
```

File: solutions/usaco-762.mdx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+<PySection>
+
+```python
+import sys
+
+sys.stdin = open("homework.in", "r")
+sys.stdout = open("homework.out", "w")
+
+input = sys.stdin.readline
```
Comment (ryanchou-dev) [inline solutions/usaco-762.mdx:24]: .

File: solutions/usaco-762.mdx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+n = int(input())
+
+questions = [*map(int, input().split())]
+
+left, right, total, minimum, ans = -1, n, [0] * n, [0] * n, []
```
Comment (ryanchou-dev) [inline solutions/usaco-762.mdx:30]: recommend defining these in sep lines

File: solutions/usaco-762.mdx (hunk 2 above)
Comment (jjliewie) [reply solutions/usaco-762.mdx:24]: okay

File: solutions/usaco-762.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-762.mdx]: yrp

File: solutions/usaco-762.mdx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+**Time Complexity:** $\mathcal{O}(N)$
+
+<LanguageSection>
+
+<PySection>
+
+```python
+with open('homework.in') as read:
+        n = int(read.readline())
+        questions = [*map(int, read.readline().split())]
```
Comment (SansPapyrus683) [inline solutions/usaco-762.mdx:21]:
```suggestion
        questions = [int(i) for i in read.readline().split()]
```

File: solutions/usaco-762.mdx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+left, right = -1, n
+total, minimum, ans = [0] * n, [0] * n, []
```
Comment (SansPapyrus683) [inline solutions/usaco-762.mdx:24]: (optional) better name for `ans`
what's the answer in the context of the problem?

File: solutions/usaco-762.mdx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+for i in range(n - 2, 0, -1):
+        cur_q = questions[i]
+
+        total[i], minimum[i] = cur_q + total[i + 1], min(minimum[i + 1], cur_q)
+
+        cur_s, tmp = total[i] - minimum[i], (n - (i + 1))
+
+        if cur_s * right > left * tmp:
+                left, right = cur_s , tmp
```
Comment (SansPapyrus683) [inline solutions/usaco-762.mdx:36]: please don't use `tmp`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-762.mdx (hunk 6 above)
Comment (jjliewie) [reply solutions/usaco-762.mdx:36]: okay

File: solutions/usaco-762.mdx (hunk 5 above)
Comment (jjliewie) [reply solutions/usaco-762.mdx]: values to obtain maximum score, so i changed it to values.

Comment (SansPapyrus683) [reply solutions/usaco-762.mdx]: `max_score_values`?

File: solutions/usaco-762.mdx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+```python
+with open('homework.in') as read:
+        n = int(read.readline())
+        questions = [int(i) for i in read.readline().split()]
+
+left, right = -1, n
+total, minimum, values = [0] * n, [0] * n, []
+
+total[n - 1], minimum[n - 1] = questions[n - 1], questions[n - 1]
```
Comment (SansPapyrus683) [inline solutions/usaco-762.mdx:26]: separate lines

File: solutions/usaco-762.mdx (hunk 5 above)
Comment (jjliewie) [reply solutions/usaco-762.mdx]: alright

File: solutions/usaco-762.mdx (hunk 7 above)
Comment (jjliewie) [reply solutions/usaco-762.mdx:26]: changing rn

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/usaco-762.mdx [hunk 8]
```diff
@@ -0,0 +1,53 @@
+---
+id: usaco-762
+source: USACO Silver 2017 December
+title: My Cow Ate My Homework
+author: Juheon Rhee
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_homework_silver_dec17.html)
```
Comment (ryanchou-dev) [inline solutions/usaco-762.mdx:8]:
```suggestion
[Official Analysis (C++ and Java)](http://www.usaco.org/current/data/sol_homework_silver_dec17.html)
```

**Review by ryanchou-dev — APPROVED** [review]
⭐

File: solutions/usaco-762.mdx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+minimum = [0] * n
+max_score_values = []
+
+total[n - 1] = questions[n - 1]
+minimum[n - 1] = questions[n - 1]
+
+for i in range(n - 2, 0, -1):
+        cur_q = questions[i]
+
+        total[i], minimum[i] = cur_q + total[i + 1], min(minimum[i + 1], cur_q)
```
Comment (SansPapyrus683) [inline solutions/usaco-762.mdx:35]: assignments on separate lines

File: solutions/usaco-762.mdx [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+        cur_s, idx = total[i] - minimum[i], (n - (i + 1))
```
Comment (SansPapyrus683) [inline solutions/usaco-762.mdx:37]: same here

**Review by envyaims — APPROVED** [review]

File: solutions/usaco-762.mdx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+        n = int(read.readline())
+        questions = [int(i) for i in read.readline().split()]
+
+left, right = -1, n
+        
+total = [0] * n
+minimum = [0] * n
+max_score_values = []
+
+total[n - 1] = questions[n - 1]
```
Comment (chezbgone) [inline solutions/usaco-762.mdx:29]: You might as well initialize `total = list(questions)` (and similarly for `minimum`.

File: solutions/usaco-762.mdx (hunk 9 above)
Comment (jjliewie) [reply solutions/usaco-762.mdx:35]: okay

File: solutions/usaco-762.mdx (hunk 10 above)
Comment (jjliewie) [reply solutions/usaco-762.mdx:37]: alright

File: solutions/usaco-762.mdx (hunk 11 above)
Comment (jjliewie) [reply solutions/usaco-762.mdx:29]: yeah but I only need the value in questions[n-1]

**Review by nwatx — APPROVED** [review]

**Pull request merged**